File: python/src/common/file.py
import os

class CommonFile():
    
    @staticmethod
    def read(path):
        print(f'filepath:{path}')
        with open(path, 'r', encoding="utf-8") as f:
            data = f.read()
            return data
        
    @staticmethod
    def write(path, content):
        with open(path, 'w', encoding="utf-8") as f:
            f.write(content)
            
    @staticmethod
    def write_add(path, content):
        with open(path, 'a', encoding="utf-8") as f:
            f.write(content)
            
    @staticmethod
    def load(path):
        print(f'filepath:{path}')
        with open(path, 'r') as f:
            data = f.read()
            return data
        
    @staticmethod
    def make(path, content, mode='x'):
        with open(path, mode) as f:
            f.write(content)
            
    @staticmethod
    def create_abs_path(path):
        if not os.path.isabs(path):
            dirname = os.path.dirname(f'{os.getcwd()}')
            file_path = f"{dirname}/{path}"
        else:
            file_path = path
        return file_path
        
    @staticmethod
    def exists_path(path):
        return os.path.exists(path) 
    
    @staticmethod
    def make_directory(path):
        os.makedirs(path)
    
    
    @staticmethod
    def get_cwd():
        return os.getcwd()
    

File: csharp/SeleniumSample/Src/Common/File.cs

using System;
using System.IO;
using System.Text;
using System.Reflection;

namespace Common
{
    class CommonFile
    {
        public static string Read(string path)
        {
            Console.WriteLine($"filepath:{path}");
            using (StreamReader sr = new StreamReader(path, Encoding.GetEncoding("utf-8")))
            {
                var data = sr.ReadToEnd();
                return data;
            }
        }

        public static void Write(string path, string content)
        {
            using (StreamWriter sw = new StreamWriter(path, false, Encoding.GetEncoding("utf-8")))
            {
                sw.WriteLine(content);
            }
        }


        public static void WriteAdd(string path, string content)
        {
            using (StreamWriter sw = new StreamWriter(path, true, Encoding.GetEncoding("utf-8")))
            {
                sw.WriteLine(content);
            }
        }

        public static string Load(string path)
        {
            Console.WriteLine($"filepath:{path}");
            using (StreamReader sr = new StreamReader(path))
            {
                var data = sr.ReadToEnd();
                return data;
            }
        }

        public static void Make(string path, string content)
        {
            using (StreamWriter sw = new StreamWriter(path, true))
            {
                sw.WriteLine(content);
            }
        }

        public static void Create(string path, string content)
        {
            using (StreamWriter sw = new StreamWriter(path, true))
            {
                sw.WriteLine(content);
            }
        }

        public static string CreateAbsPath(string path)
        {
            if (System.IO.Path.IsPathRooted(path))
            {
                return path;
            }
            Assembly? myAssembly = Assembly.GetEntryAssembly();
            var dirName = System.IO.Path.GetDirectoryName(myAssembly?.Location);
            var filePath = $"{dirName}/{path}";
            return filePath;
        }

        public static bool ExistsPath(string path)
        {
            return File.Exists(path);
        }

        public static bool ExistsDirecoryPath(string path)
        {
            return Directory.Exists(path);
        }

        public static void CreateDirectory(string path)
        {
            Directory.CreateDirectory(path);
        }

        public static string GetCwd()
        {
            Assembly? myAssembly = Assembly.GetEntryAssembly();
            return System.IO.Path.GetDirectoryName(myAssembly?.Location ?? "") ?? "";
        }

    }
}